R/models.R

```diff
@@ -11,15 +11,19 @@
 #'   \item{min.node.size: Minimal node size.}
 #' }
 #'
-#' \strong{k-nearest neighbors}: "knn". Regression and classification.
-#' Implemented via \code{kknn}.
+#' \strong{XGBoost}: "xgb". eXtreme Gradient Boosting
+#' Implemented via \code{xgboost}.
 #' \itemize{
-#'   \item{kmax: Number of neighbors to consider.}
-#'   \item{distance: Minkowsky distance parameter, (0, Inf). 1 = Manhatten, 2 =
-#'   Euclidian, -> Inf = Chebyshev.}
-#'   \item{kernal: Kernal to use. Possible choices are "rectangular" (standard
-#'   knn), "triangular", "epanechnikov", "biweight", "triweight", "cos", "inv",
-#'   "gaussian", "rank", or "optimal".}
+#'   \item{eta: Control for learning rate, [0, 1]}
+#'   \item{gamma: Threshold for further cutting of leaves, [0, Inf].
+#'   Larger is more conservative.}
+#'   \item{max_depth: Maximum tree depth, [0, Inf]. Larger means more complex
+#'   models and so greater likelihood of overfitting. 0 produces no limit on depth.}
+#'   \item{subsample: Fraction of data to use in each training instance, (0, 1].}
+#'   \item{colsample_bytree: Fraction of features to use in each tree, (0, 1].)}
+#'   \item{min_child_weight: Minimum sum of instance weight need to keep partitioning,
+#'   [0, Inf]. Larger values mean more conservative models.}
+#'   \item{nrounds: Number of rounds of boosting, [0, Inf)}
 #' }
```

Comment:
It would be great to have a comment about relative model training time in the `machine_learn` docs to give people a ballpark. Something like, GLM is probably fastest, RF is next, xgb is slowest. This could also live in the `models` file.

Reply:
Good idea. I'll do that in #1107